types/actions.ts

```diff
@@ -13,3 +13,13 @@ export function isSuccess(result: ActionResult): result is SuccessResult {
 export function isError(result: ActionResult): result is ErrorResult {
     return Boolean((result as ErrorResult).error);
 }
+
+export type Previous = {
```

Comment:
Thanks for adding the type :)
I don't think each field needs to be optional however.
If we ever needed that case, we could always use `Partial<Previous>`